sample/petstore-vertx-rx-server/src/main/java/io/swagger/server/api/verticle/PetApiImpl.java

```diff
@@ -0,0 +1,70 @@
+package io.swagger.server.api.verticle;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.swagger.server.api.model.Category;
+import io.swagger.server.api.model.ModelApiResponse;
+import io.swagger.server.api.model.Pet;
+import io.swagger.server.api.model.Pet.StatusEnum;
+import rx.Single;
+import rx.functions.Func1;
+
+public class PetApiImpl implements PetApi {
+
+    @Override
+    public Single<Void> addPet(Pet body) {
+        return Single.just(null);
+    }
+
+    @Override
+    public Single<Void> deletePet(Long petId, String apiKey) {
+        return Single.just(null);
+    }
+
+    @Override
+    public Single<List<Pet>> findPetsByStatus(List<String> status) {
+        List<Pet> pets = new ArrayList<>();
+        pets.add(new Pet(1L, new Category(1L, "dog"), "rex", new ArrayList<>(), new ArrayList<>(), StatusEnum.AVAILABLE));
+
+        return Single.just(pets);
+    }
+
+    @Override
+    public Single<List<Pet>> findPetsByTags(List<String> tags) {
+        return Single.just(null);
+    }
+
+    @Override
+    public Single<Pet> getPetById(Long petId) {
+        return Single.just(petId).flatMap(new Func1<Long, Single<Pet>>() {
+            @Override
+            public Single<Pet> call(Long id) {
+                if (id.equals(1L))
+                    return Single.just(new Pet(1L, new Category(1L, "dog"), "rex", new ArrayList<>(), new ArrayList<>(), StatusEnum.AVAILABLE));
+                else if (id.equals(2L))
+                    return Single.error(new NullPointerException("simulation"));
+                else
+                    return Single.error(PetApiException.Pet_getPetById_404_Exception);
+            }
+        });
+
+    }
+
+    @Override
+    public Single<Void> updatePet(Pet body) {
+        return Single.just(null);
```

Comment:
In rxjava2, null is no longer an allowed item to emit in an observable stream.

That being said, I think vertx-rx is currently depending on rxjava 1.x. In this case, your code is correct, but will break at some point.

The proper way to do this in rxjava2 would be to use Completable, I think: http://reactivex.io/RxJava/2.x/javadoc/io/reactivex/Completable.html

I think Completable is also present in rxjava 1.x. If so, I recommend to use that instead of `Single<Result>` in cases where there is no result.